devito/ops/transformer.py

```diff
@@ -169,12 +172,18 @@ def create_ops_dat(f, name_to_ops_dat, block):
             ops_dat_array,
             ListInitializer(dat_decls)
         )))
+
+        # Inserting the ops_dat array in case of TimeFunction.
+        name_to_ops_dat[f.name] = ops_dat_array
+
     else:
         ops_dat = OpsDat("%s_dat" % f.name)
         name_to_ops_dat[f.name] = ops_dat
 
-        d_p_val = tuple(sympify([p[0] + h[0] for p, h in zip(f.padding, f.halo)]))
-        d_m_val = tuple(sympify([-(p[1] + h[1]) for p, h in zip(f.padding, f.halo)]))
+        d_p_val = tuple(sympify([p[0] + h[0]
```

Comment:
Actually I also don't know.
My guess is that the padding and halo attribute are both python `int` at this point, but we need this information to be translated back to `Devito` API so it can use it to generate nodes that will declare the `d_p` and `d_m` variables.

For example, it need to generate something like this in C:
`int v_d_p[2] = {2, 2};`

But @vincepandolfo can explain better than me this part.

Reply:
Also, this part shouldn't be changed in my commit. 
It was just the format but the previous format was already good. I think my IDE changed it and I didn't realize.

Reply:
I explained it in the comment above . Let me know if you got it

Reply:
We need `simpify` as otherwise `simpy` will complain later. I do not remember the details, though. Should be easy enough to try it out

Reply:
Do not use magic numbers 0/1 -- use `_size_halo.left` for example